src/rust/engine/process_execution/src/speculate.rs

```diff
@@ -0,0 +1,214 @@
+use std::time::{Duration, Instant};
+use std::sync::Arc;
+use tokio_timer::Delay;
+use futures::future::{Either, Future, result};
+use boxfuture::{BoxFuture, Boxable};
+use workunit_store::WorkUnitStore;
+use super::{
+  CommandRunner,
+  ExecuteProcessRequest,
+  FallibleExecuteProcessResult
+};
+
+
+#[derive(Clone)]
+pub struct SpeculatingCommandRunner {
+  primary: Arc<dyn CommandRunner>,
+  secondary: Arc<dyn CommandRunner>,
+  speculation_timeout_ms: u64,
```

Comment:
Any reason to take a `u64` here rather than a `Duration`?

Reply:
ignorance mostly :)